source/code/providers/Container_HostInventory_Class_Provider.cpp

```diff
@@ -0,0 +1,212 @@
+                string volumeList;
+                if(volumeEntry != NULL)
+                {
+                    for(int i =0; i < cJSON_GetArraySize(volumeEntry);i++)
+                    {
+                        volumeList.append(string(cJSON_GetArrayItem(volumeEntry, i)->valuestring));
+                    }
+                }
+                hostInventoryInstance.Volume_value(volumeList.c_str());
+                //Get network info
+                cJSON* networkEntry = cJSON_GetObjectItem(plugins_entry, "Network");
+                string networkList;
+                if(networkEntry != NULL)
+                {
+                    for(int i =0; i < cJSON_GetArraySize(networkEntry);i++)
+                    {
+                        networkList.append(string(cJSON_GetArrayItem(networkEntry, i)->valuestring));
+                        networkList.append(" ");
+                    }
+                }
+                hostInventoryInstance.Network_value(networkList.c_str());
+            }
+
+            //Node role is obtained from the computer name. e.g. k8s-master-71E8D996-0 => master role
+            if(computerName.find("master") != string::npos || computerName.find("manager") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.NodeRole_value("Master");  
+            }
+            else if(computerName.find("agent") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.NodeRole_value("Agent");  
+            }
+            else
+            {
+                hostInventoryInstance.NodeRole_value("Not Orchestrated");  
+            }   
+
+            //Orchestrator type is obtained from the node name. example of node names swarm-manager-vmss_0 for swarm, k8s-master-71E8D996-0 for kubernetes
+            if(computerName.find("swarmm") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("Swarm Mode");  
+            }
+            else if(computerName.find("swarm") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("Swarm");  
+            }
+            else if(computerName.find("k8") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("Kubernetes");  
+            }
+            else if(computerName.find("dcos") != string::npos)
+            {
+                //swarm
+                hostInventoryInstance.OrchestratorType_value("DC/OS");  
+            }
+            else
+            {
+                hostInventoryInstance.OrchestratorType_value("None");  
+            }   
+                        
+            // Clean up object
+            cJSON_Delete(response[0]);
+        }
+        else
+        {
+            syslog(LOG_WARNING, "API call in Container_HostInventory::InspectHost to get host info returned null");
```

Comment:
So is syslog your logging mechanism, vs. the PAL logging facilities? If so, you should use that for the `printf` statement above.